src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -66,768 +67,819 @@ parseAuthorizeCredentials(Json::Value const& jv)
-            }
-            else
-            {
-                jvResult[jss::error] = "malformedRequest";
-            }
+            jvResult[jss::error] = "malformedRequest";
+            return std::nullopt;
         }
-        else if (context.params.isMember(jss::mptoken))
+        return uNodeIndex;
+    }
+
+    if (!mptJson.isMember(jss::mpt_issuance_id) ||
+        !mptJson.isMember(jss::account))
+    {
+        jvResult[jss::error] = "malformedRequest";
+        return std::nullopt;
+    }
+    try
+    {
+        auto const mptIssuanceIdStr = mptJson[jss::mpt_issuance_id].asString();
+
+        uint192 mptIssuanceID;
+        if (!mptIssuanceID.parseHex(mptIssuanceIdStr))
+            Throw<std::runtime_error>("Cannot parse mpt_issuance_id");
+
+        auto const account =
+            parseBase58<AccountID>(mptJson[jss::account].asString());
+
+        if (!account || account->isZero())
         {
-            expectedType = ltMPTOKEN;
-            auto const& mptJson = context.params[jss::mptoken];
-            if (!mptJson.isObject())
-            {
-                if (!uNodeIndex.parseHex(mptJson.asString()))
-                {
-                    uNodeIndex = beast::zero;
-                    jvResult[jss::error] = "malformedRequest";
-                }
-            }
-            else if (
-                !mptJson.isMember(jss::mpt_issuance_id) ||
-                !mptJson.isMember(jss::account))
-            {
-                jvResult[jss::error] = "malformedRequest";
-            }
-            else
-            {
-                try
-                {
-                    auto const mptIssuanceIdStr =
-                        mptJson[jss::mpt_issuance_id].asString();
-
-                    uint192 mptIssuanceID;
-                    if (!mptIssuanceID.parseHex(mptIssuanceIdStr))
-                        Throw<std::runtime_error>(
-                            "Cannot parse mpt_issuance_id");
-
-                    auto const account = parseBase58<AccountID>(
-                        mptJson[jss::account].asString());
-
-                    if (!account || account->isZero())
-                        jvResult[jss::error] = "malformedAddress";
-                    else
-                        uNodeIndex =
-                            keylet::mptoken(mptIssuanceID, *account).key;
-                }
-                catch (std::runtime_error const&)
-                {
-                    jvResult[jss::error] = "malformedRequest";
-                }
-            }
+            jvResult[jss::error] = "malformedAddress";
+            return std::nullopt;
         }
-        else
+
+        return keylet::mptoken(mptIssuanceID, *account).key;
+    }
+    catch (std::runtime_error const&)
+    {
+        jvResult[jss::error] = "malformedRequest";
+        return std::nullopt;
+    }
+}
+
+using FunctionType =
+    std::function<std::optional<uint256>(Json::Value const&, Json::Value&)>;
```

Comment:
I don't think you need runtime polymorphism here.
```suggestion
using FunctionType =
    std::optional<uint256>(*)(Json::Value const&, Json::Value&);
```

Reply:
I'm unfamiliar with the pros and cons of `std::function` vs C-style pointers, I just assumed the norm was the more modern option.